### ISSUE TITLE
ipympl figures not shown in Firefox (works fine with Chrome)

### ISSUE THREAD

**Issue by paugier** [voila-dashboards/voila#693]
It's sad... Firefox 80.0

I reproduce with voila/notebooks/ipympl.ipynb

Voila, ipympl and all packages installed in an empty environment today with conda.

The console in Firefox says `Uncaught (in promise) DOMException: The object could not be cloned.`

I tried with `voila ipympl.ipynb --debug`. The log ends by
```
WARNING:tornado.general:404 GET /api/kernels/d5b19400-beaf-434c-9053-a0ad52b539fe/channels?session_id=2a60a4bb-36f4-49c0-9f7a-a8e6f34f5300 (127.0.0.1): Kernel does not exist: d5b19400-beaf-434c-9053-a0ad52b539fe
[Voila] Using contents: services/contents
WARNING:tornado.access:404 GET /api/kernels/d5b19400-beaf-434c-9053-a0ad52b539fe/channels?session_id=2a60a4bb-36f4-49c0-9f7a-a8e6f34f5300 (127.0.0.1) 36.14ms
[Voila] Initializing websocket connection /api/kernels/d5b19400-beaf-434c-9053-a0ad52b539fe/channels
[Voila] WARNING | Replacing stale connection: d5b19400-beaf-434c-9053-a0ad52b539fe:2a60a4bb-36f4-49c0-9f7a-a8e6f34f5300
[Voila] Initializing websocket connection /api/kernels/d5b19400-beaf-434c-9053-a0ad52b539fe/channels
[Voila] WARNING | Replacing stale connection: d5b19400-beaf-434c-9053-a0ad52b539fe:2a60a4bb-36f4-49c0-9f7a-a8e6f34f5300
```

**Comment by paugier** [comment]
It is really a very annoying issue that Voilà + ipympl do not work with recent versions of Firefox. 

Note that it works fine with Firefox + Jupyter Notebook + ipympl.

**Comment by paugier** [comment]
Ah interesting, it works fine with voila 0.1.21, as in https://github.com/voila-dashboards/voila/blob/master/environment.yml.

But not with  0.1.22,  0.1.23 and master.

**Comment by martinRenou** [comment]
I see no problem with latest ipympl, voila 0.1.23, and Firefox 80.0.1 (64-bit). Which Firefox version do you have?

Could you paste a bit more detail about the `Uncaught (in promise) DOMException: The object could not be cloned`? It seems like the interesting part, in which file was this triggered?

**Comment by paugier** [comment]
Strange. I also use Firefox 80.0.1 (64-bit) (with ubuntu 20.04).

The error is triggered from `(index)`, l.31, second line of the block:

```
     console.error = (...args) => {
        window.top.postMessage({ level: "error", msg: ["[VOILA]:", ...args] });
        _error(...args);
    };
```

I launch voila with: `(venv_voila) pierre@gre ~/Dev/voila/notebooks master $ voila ipympl.ipynb` and in the virtual env `venv_voila`, there is:

```
$ pip freeze               
argon2-cffi==20.1.0
async-generator==1.10
attrs==20.2.0
backcall==0.2.0
bleach==3.1.5
certifi==2020.6.20
cffi==1.14.2
cycler==0.10.0
decorator==4.4.2
defusedxml==0.6.0
entrypoints==0.3
ipykernel==5.3.4
ipympl==0.5.8
ipython==7.18.1
ipython-genutils==0.2.0
ipywidgets==7.5.1
jedi==0.17.2
Jinja2==2.11.2
jsonschema==3.2.0
jupyter-client==6.1.7
jupyter-core==4.6.3
jupyter-server==0.1.1
jupyterlab-pygments==0.1.1
kiwisolver==1.2.0
MarkupSafe==1.1.1
matplotlib==3.3.2
mistune==0.8.4
nbclient==0.5.0
nbconvert==5.6.1
nbformat==5.0.7
nest-asyncio==1.4.0
notebook==6.1.4
numpy==1.19.2
packaging==20.4
pandocfilters==1.4.2
parso==0.7.1
pexpect==4.8.0
pickleshare==0.7.5
Pillow==7.2.0
prometheus-client==0.8.0
prompt-toolkit==3.0.7
ptyprocess==0.6.0
pycparser==2.20
Pygments==2.7.0
pyparsing==2.4.7
pyrsistent==0.17.3
python-dateutil==2.8.1
pyzmq==19.0.2
Send2Trash==1.5.0
six==1.15.0
terminado==0.8.3
testpath==0.4.4
tornado==6.0.4
traitlets==5.0.4
voila==0.1.23
wcwidth==0.2.5
webencodings==0.5.1
widgetsnbextension==3.5.1
```

**Comment by TheoMathurin** [comment]
I'm facing the same issue on Firefox 68.12.0esr and 80.0.1. Works fine on Chromium.

**Comment by paugier** [comment]
I'm also get the issue with Firefox 68.12.0esr (on Debian 10, i.e. Debian stable). For example just by clicking on the Binder link in the README (because `environment.yaml` was updated in https://github.com/voila-dashboards/voila/commit/44f8deb7ea31462dc3049796d1f5e399073a48cc so it stopped using Voila 0.1.21).

**Comment by paugier** [comment]
I don't know if can be useful, but Firefox says
```
DataCloneError: The object could not be cloned. 4 ipympl.ipynb:30
    error ipympl.ipynb:30
    exports host-report-errors.js:6
    q es.promise.js:131
    n es.promise.js:143
    exports perform.js:3
    U es.promise.js:140
    f setImmediate.js:40
    f setImmediate.js:69
    a setImmediate.js:109
```

**Comment by martinRenou** [comment]
I suspect this line to be the culprit: https://github.com/voila-dashboards/voila/blame/ae3fe3d1f1ea44e2ef32e4d9f94283d3861bf431/js/src/manager.js#L102

Would you mind creating a dev install of Voila, removing this line, see if that fixes your issue?

The proper fix might not be to remove this line though, but it might give us a bit more information.

**Comment by paugier** [comment]
Yes it works when this line is commented! Now I feel like a JS dev!

**Comment by TheoMathurin** [comment]
Well... in my case with a dev install, it worked the first time _without_ having removed line 102 in manager.js. Now the problem has resurfaced, after several tries using fresh dev installs with or without commenting out the line.

**Comment by martinRenou** [comment]
Maybe it's an installation issue? You need to rebuild the JS after making changes

**Comment by TheoMathurin** [comment]
Indeed, I have rebuilt it and even tried with a new clone of the voila repository. I'll continue to investigate and let you know, but the weird thing is that at first it worked while I had not commented out that line in manager.js.

**Comment by martinRenou** [comment]
Thank you for investigating! I would do it myself but I cannot reproduce at all.

**Comment by paugier** [comment]
I confirm that it is fixed on master (with Firefox 78.3.0esr (64 bits))!

Do you plan to release a new version soon? :slightly_smiling_face:
